app/api/routes.py

```diff
@@ -174,3 +181,20 @@ def set_resource(id, json, db):
             return jsonify(resource.serialize)
         else:
             return jsonify({})
+
+
+def create_resource(json, db):
+    langs, categ = get_attributes(json)
+    new_resource = Resource(
+        name=json.get('name'),
+        url=json.get('url'),
+        category=categ,
+        languages=langs,
+        paid=json.get('paid'),
+        notes=json.get('notes'))
+
+    try:
+        db.session.add(new_resource)
+    except Exception as e:
```

Comment:
The data isn't getting saved in the database, you need to add a call to `db.session.commit()` in order to save it. And when you add that, please also add a call to `db.session.rollback()` in the exception handler so that if anything goes wrong, the db stays in a consistent state. Should look like this:
```python
    try:
        db.session.add(new_resource)
        db.session.commit()
    except Exception as e:
        db.session.rollback()
        print('exception', e)
```